### ISSUE TITLE
'gs repo sync' error for on-premise GitLab instances

### ISSUE THREAD

**Issue by gr4cza** [abhinav/git-spice#391]
`gs repo sync` returns 
```
INF main: already up-to-date
ERR Could not guess repository from remote URL url=https://gitlab.xyz.io/abc.git
ERR Are you sure the remote identifies a supported Git host?
FTL gs: unsupported Git remote URL
``` 
for on-premise GitLab instances. It would be really nice, if it could handle custom urls without errors.

**Comment by abhinav** [comment]
Hello! Thanks for reporting. This is worth fixing.

Note that without the ability to check for merged CRs (without #348), `gs repo sync` will probably just do `git pull`, but it's a reasonable graceful degradation.
There may be some room for local merge detection, but I suspect it won't handle cases like squash-merged CRs.

Just out of curiosity: Are you using `submit` with `--no-publish` (or the [spice.submit.publish](https://abhinav.github.io/git-spice/cli/config/#spicesubmitpublish) configuration) to manage the stack?

**Comment by gr4cza** [comment]
Hi! First of all, thank you for your amazing work!

Yes, i'm using it with the `spice.submit.publish` flag set to `false`. It is really handy to force push multiple branches at the same time.
